09-reproducible-data.Rmd

```diff
@@ -1 +1,83 @@
+
+One way to easily handle different package versions is with the {renv} package. This package allows researchers to set the versions for each package used and easily manage package dependencies. Specifically, {renv} creates isolated, project-specific environments that record the packages and their versions used in the code. When initiated by a new user, {renv} checks whether the installed packages are consistent with the recorded version for the project. If not, it installs the correct versions and ensures that others can replicate the project's environment to rerun the code and obtain consistent results.
+
+## Workflow Management: {targets}
+
+With complex studies involving multiple code files, steps, and dependencies it can be important to ensure that all pieces are implemented in the correct order. This could be done manually (e.g., numbering files to indicate the order or providing detailed documentation on the order) or we can automate the process to ensure the code flows sequentially. Ensuring that the code is run in the correct order helps ensure that the research is reproducible and that anyone can easily pick up the code and know exactly how to run the code to get the same results.
+
+To automate, we recommend using the {targets} package, a workflow manager enabling us to document, automate, and execute complex data workflows with multiple steps and dependencies. With this package, we first define the order of execution for our code, and then it will consistently execute the code in that order each time it is run. One nice feature of {targets} is that if you change code later in the workflow, only the affected code and its downstream targets (i.e., the subsequent code files) are re-executed when we change a script. The {targets} package also provides interactive progress monitoring and reporting, allowing us to track the status and progress of our analysis pipeline. 
+
+## Documentation: Quarto/R Markdown
+
+Documenting and describing decisions help other researchers reproduce your work and findings. Methodology reports can be one way to do this, providing a significant document of all project decisions. Additionally, using tools like Quarto and R Markdown allow us to create documents that integrate code and text. These documents present analysis results alongside the report's narrative, so there's no need to copy and paste code output into the final documentation. By eliminating manual steps, we can reduce the chances of errors in the final output. 
+Quarto and R Markdown documents also allow users to re-execute the underlying code when needed. Therefore, another team member can easily recreate the report and obtain the same results. If outputting the file as HTML, these tools can also allow for more robust integration of interactive plots, images, videos, links, and anything else researchers might need to document for a project. 
```

Comment:
Since HTML is mentioned, I think we should also mention that Quarto and R Markdown allow for output in various different formats, each with their own benefits.

Reply:
Or, that might not be necessary since it's not really related to reproducibility. I think this might have come from the original communicating results chapter.

I would say there are two main ways that Quarto and R Markdown contribute to reproducibility:

1. Others can see steps taken and follow the code, reproducing the results
2. Fewer manual steps, such as no copy/paste